src/game/room.c

```diff
@@ -162,27 +177,13 @@ static void Room_RemoveFlipItems(ROOM_INFO *r)
 int16_t Room_GetTiltType(
     const SECTOR_INFO *sector, int32_t x, int32_t y, int32_t z)
 {
-    ROOM_INFO *r;
+    sector = Room_GetPitSector(sector, x, z);
 
-    while (sector->pit_room != NO_ROOM) {
-        r = &g_RoomInfo[sector->pit_room];
-        sector = &r->sectors
-                      [((z - r->z) >> WALL_SHIFT)
-                       + ((x - r->x) >> WALL_SHIFT) * r->z_size];
-    }
-
-    if ((y + 512) < sector->floor.height) {
+    if ((y + STEP_L * 2) < sector->floor.height) {
```

Comment:
Maybe after the refactors we can combine the `walkable` function check here. We should have better floordata structure that maybe will allow us to easily add another if check on whether to return no tilt.